src/helpers/navigation.ts

```diff
@@ -0,0 +1,69 @@
+import { ContentFilter } from '../models/ContentFilter';
+import { Feed } from '../models/Feed';
+import { SubCategory, NewsSource } from '../models/recommendation/NewsSource';
+import { LocalFeed } from '../social/api';
+
+export interface Routes {
+    App: {};
+    Loading: {};
+    Welcome: {};
+    Post: {};
+    Root: {};
+    ProfileTab: {};
+    PostTab: {};
+    FavoriteTab: {};
+    SettingsTab: {};
+    AllFeedTab: {};
+    BugReportView: {};
+    SwarmSettingsContainer: {};
+    FilterListEditorContainer: {};
+    Backup: {};
+    Restore: {};
+    BackupRestore: {};
+    LogViewer: {};
+    Debug: {};
+    FeedListViewerContainer: {
+        showExplore: boolean,
+        feeds?: Feed[],
+    };
+    Feed: {
+        feedUrl: string,
+        name: string,
+    };
+    FeedInfo: {
+        feed: Feed;
+    };
+    EditFilter: {
+        filter: ContentFilter,
+    };
+    FeedSettings: {
+        feed: LocalFeed,
+    };
+    FeedFromList: {
+        feedUrl: string,
+        name: string,
+    };
+    CategoriesContainer: {
+
+    };
+    SubCategoriesContainer: {
+        title: string,
+        subCategories: SubCategory[],
+    };
+    NewsSourceFeed: {
+        feed: Feed,
+    };
+    NewsSourceGridContainer: {
+        newsSources: NewsSource[],
+        subCategoryName: string,
+    };
+    YourTab: {};
+}
+
+export interface TypedNavigation {
+    goBack: (routeKey?: string | null) => boolean;
+    navigate: <T, K extends keyof T>(routeKey: K, params: T[K]) => boolean;
```

Comment:
This is quite verbose so I checked and everywhere we use this it is used with `T = Routes`. Therefore we could use this instead:
`navigate: <K extends keyof Routes>(routeKey: K, params: Routes[K]) => boolean;
`
Then using it would be as simple as:
`navigation.navigate('Loading', {});
`

Reply:
👍

Reply:
it would be nice to get rid of the {} as well, because that allows passing anything. (in case it is an empty object in the Routes)

Reply:
Well, that's difficult to do in Typescript, because structurally every interface is equivalent with the empty interface.